docs/README-METRICS.md

```diff
@@ -0,0 +1,165 @@
+disabling tests, and secondly, to collect user feedback.  Our main system
+health measures will be the number of tests installed over time and the count
+of user feedback recorded over time.  These will allow us to quickly see
+disruptions in service regardless of cause.
+
+![](metrics-feedback-recorded.png)
+![](metrics-total-installs.png)
+
+# Data Collection Details
+
+There are two sources for metrics data - the server and the client.
+
+**The server** will use server logs to store metric data which will be picked
+up by the measurement team ([some docs][3]), parsed and transformed if
+necessary, and pushed into ElasticSearch (as scale permits), S3, and Redshift.
+Events recorded in the logs will include:
+
+* request.summary (a generic event fired for each visit)
+* testpilot.newuser
+* testpilot.newfeedback
+* testpilot.main-install
+* testpilot.test-install
+
+Data recorded by the log files will include standard system logging (in the
+example below, this is everything down to the blank line in the Fields object)
+as well as:
+* Feature Switch Set (if applicable)
+* Campaign (marketing identifier) (if applicable)
+
+Using the [standard format][4] an example object will look like:
+
+```js
+{
+  "Timestamp": 1457462527,
+  "Type": "testpilot.test-install",
+  "Logger": "TestPilotWeb",
+  "Hostname": "testpilot.servers.s3",
+  "EnvVersion": "2.0",  // We're conforming to the Fields spec
+  "Severity": 0,
+  "Pid": 1000,
+  "Fields":{
+    "agent": "User Agent String",
+    "path": "/tests/",
+    "method": "GET",
+    "code": 200,
+    "errno": 0,
+    "lang": "en-US",
+    "uid": 1000,  // This is the Test Pilot UID
+    "service": "universal_search@mozilla",  // If the ping comes from a test within the program, the identifier goes here
+    "context": "",
+    "msg": "New Test Installed: Universal Search",
+    "remoteAddressChain": ["1.2.3.4", "4.3.2.1"],
+    "rid": "0b65c6fb-26d5-4367-be29-b3094f0c69d5",
+    "t": 256,
+
+    "feature_switches": "",
+    "campaign": ""
+  }
+}
+```
+
+The second source, **the client**, will use the existing Telemetry system (the
+system we use for measuring all kinds of things in Firefox, [see examples][5])
+which has a [submitExternalPing() function][6].  This function will allow us to
+submit arbitrary payloads to the Telemetry pipeline which are then scrubbed and
+put into an appropriate Redshift cluster with all our server log data.  The
+client pings will adhere to the [common ping format][7], will include the
+clientId and [environment][8], and will have as a payload:
+
+* A periodic (every 24 hours) ping which includes:
+  * For each test:
+    * enabled/disabled state with timestamp of last toggle
+    * feature switch status
+  * User Agent
+  * Test Pilot User ID
+
+An example payload (within the full ping) would look like:
+```js
+{"tests":
+  {"universal_search":
+    {"last_enabled": 1457462200,
+     "last_disabled": 1457461100,
```

Comment:
This doesn't have the "state" as mentioned above in the docs. Unless it is assumed that it is derived from the latest timestamp?